src/sys/signal.rs

```diff
@@ -388,8 +388,12 @@ pub fn pthread_sigmask(how: SigFlags,
     Errno::result(res).map(drop)
 }
 
-pub fn kill(pid: libc::pid_t, signal: Signal) -> Result<()> {
-    let res = unsafe { libc::kill(pid, signal as libc::c_int) };
+pub fn kill(pid: libc::pid_t, signal: Option<Signal>) -> Result<()> {
```

Comment:
What about us supporting Rust (1.7+), though? If someones crate wants to continue to support 1.7+, it has to change its source (which I don't have a problem with in the first place).

So asked differently: Are you saying, that we will passively take advantage of the new impl on new enough Rust versions anyway?